### PR TITLE
feat(submodel registry): Establish submodel registry

### PR DESCRIPTION
**Pull request by NilsXitaso**

### CODE REVIEW

File: src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx [hunk 1]
```diff
@@ -4,35 +4,45 @@ import { useIsMobile } from 'lib/hooks/UseBreakpoints';
...
+    const env = useEnv();
 
     useAsyncEffect(async () => {
         if (!props.smReferences) return;
 
         const submodels: { id: string; label: string; metadata?: Submodel; endpoint?: string }[] = [];
 
+        async function fetchSubmodelFromRepo(reference: Reference) {
+            const id = reference.keys[0].value;
+            const metadata = await submodelClient.getSubmodelMetaDataById(id);
```
Comment (hofermo) [inline src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx:42]: Unsure if my PR #30 will change this as well, please let us check this when you merge

File: src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx [hunk 2]
```diff
@@ -4,35 +4,45 @@ import { useIsMobile } from 'lib/hooks/UseBreakpoints';
...
+            submodels.push({ id, label: metadata.idShort ?? '', metadata });
+        }
```
Comment (hofermo) [inline src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx:44]: Why no try catch as below?

File: src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx [hunk 3]
```diff
@@ -75,12 +90,19 @@ export function SubmodelsOverviewCard(props: SubmodelsOverviewCardProps) {
         let fetchedSubmodel;
 
         if (selectedSubmodel) {
-            if (registryAasData && selectedSubmodel.endpoint) {
-                fetchedSubmodel = await getSubmodelFromSubmodelDescriptor(selectedSubmodel.endpoint);
+            if (selectedSubmodel.endpoint) {
+                try {
+                    fetchedSubmodel = await getSubmodelFromSubmodelDescriptor(selectedSubmodel.endpoint);
+                } catch (e) {
+                    // expexted behaviour if submodel registry is not available or submodel is not found there
```
Comment (hofermo) [inline src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx:97]: A catch with no logic?

File: src/app/[locale]/viewer/_components/submodel-elements/entity-component/RelationShipDetailsDialog.tsx [hunk 4]
```diff
@@ -24,17 +25,28 @@ export function RelationShipDetailsDialog(props: RelationShipDetailsModalProps)
...
-                    submodelRefs.map((ref) => submodelClient.getSubmodelById(ref.keys[0].value)),
-                );
+                const submodels = [] as Submodel[];
+                
+                for (const reference of submodelRefs) {
+                    const id = reference.keys[0].value;
+                    try {
+                        const submodelFromRegistry = env.SUBMODEL_REGISTRY_API_URL ? await submodelRegistryServiceClient.getSubmodelDescriptorsById(reference.keys[0].value) : null
+                        submodels.push(submodelFromRegistry);
+                    } catch (e) {
```
Comment (hofermo) [inline src/app/[locale]/viewer/_components/submodel-elements/entity-component/RelationShipDetailsDialog.tsx:44]: same here

**Review by hofermo — CHANGES_REQUESTED** [review]

**Review by hofermo — CHANGES_REQUESTED** [review]
see my comments

**Review by hofermo — APPROVED** [review]

**Review by GailMelanie — APPROVED** [review]

**Pull request merged**